charts/neon-proxy/values.yaml

```diff
@@ -4,6 +4,13 @@
 
 replicaCount: 1
 
+# -- strategy override for deployment
+deploymentStrategy:
+  type: Recreate
+
+# -- Deployment's terminationGracePeriodSeconds
+terminationGracePeriodSeconds: 30
```

Comment:
Yes, if we want to set it to something else than 30. Currently it's 30s even thought it's not explicitly in the template.

Reply:
Will you push the PR to the neon with the updated value for proxy as well? (with 1 week value as we discussed)

Reply:
hmm, it seems that we have the wrong expectation of this flag. The main idea, is not blocking other releases, but allows existing proxies live for some time

Reply:
> No way set "days" or "weeks" for graceful period. It will block deploy for this time.
It won't block, because we'll set the strategy to RollingUpdate and allow large enough MaxSurge. So we can set it to a large value (e.g 7*24*60*60) and have deployments working just fine. The old ones will sit in `Terminating` until `terminationGracePeriodSeconds` has passed (and also the sleep we put to `preStop` hook).

Reply:
`helm upgrade` command will run for weeks (and GA runner should support that) and no new rollouts during this rollout as helm chart state will be `pending-rollout`

Reply:
I'm pretty sure `--wait` passes once `MaxUnavailable` is satisfied, as it would watch `Deployment`'s status.